HADOOP-19540. [JDK17] Add ubuntu:noble as a build platform with JDK-17 as default

vinayakumarb opened the pull request:
### Description of PR
1. Added "ubuntu:noble" to supported docker platforms.
2. Modified `start-build-env.sh` to use expected Dockerfiles dynamically. ex: `bash start-build-env.sh ubuntu_24` where `ubuntu_24` is the extra suffix in the name of the Dockerfile.
3. Added `jdk17` profile activated when jdk17 is available.
4. Had to add `#include <stdint.h>` to header files which misses it to successfully compile with g++ (13.3) in ubuntu 24.

### How was this patch tested?
Successfully built the entire Hadoop tar with native support on env created using `bash start-build-env.sh ubuntu_24`

### For code changes:

- [y] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [NA] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [NA] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [NA] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

ayushtkn reviewed (commented):
The error message on aarch64 ain't cheese
```
ayushsaxena@ayushsaxena hadoop % ./start-build-env.sh ubuntu_24 
[+] Building 0.0s (2/2) FINISHED                                                                                                                                                              
 => [internal] load .dockerignore                                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                                          0.0s
 => [internal] load build definition from Dockerfile_ubuntu_24_aarch64                                                                                                                   0.0s
 => => transferring dockerfile: 2B                                                                                                                                                       0.0s
ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount1675973539/Dockerfile_ubuntu_24_aarch64: no such file or directory
ayushsaxena@ayushsaxena hadoop % 
```

Can we update the BUILDING.txt to reflect which ones are supported for aarch64. 

Rest if the build comes clean, changes LGTM

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  45m 34s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  hadolint  |   0m  1s |  |  hadolint was not available.  |
| +0 :ok: |  shellcheck  |   0m  1s |  |  Shellcheck was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +0 :ok: |  jsonlint  |   0m  1s |  |  jsonlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 16s |  |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |  36m 34s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/branch-mvninstall-root.txt) |  root in trunk failed.  |
| -1 :x: |  compile  |  13m  0s | [/branch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/branch-compile-root.txt) |  root in trunk failed.  |
| -1 :x: |  mvnsite  |   4m 48s | [/branch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/branch-mvnsite-root.txt) |  root in trunk failed.  |
| -1 :x: |  javadoc  |   6m 47s | [/branch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/branch-javadoc-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  shadedclient  |  45m  6s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |  35m 57s | [/patch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/patch-mvninstall-root.txt) |  root in the patch failed.  |
| -1 :x: |  compile  |  13m  2s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  cc  |  13m  2s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  golang  |  13m  2s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  javac  |  13m  2s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -1 :x: |  mvnsite  |   4m 59s | [/patch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/patch-mvnsite-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| -1 :x: |  javadoc  |   6m 29s | [/patch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/patch-javadoc-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  shadedclient  |  45m 57s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 598m  9s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 33s |  |  The patch does not generate ASF License warnings.  |
|  |   | 855m 27s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.util.TestNativeCodeLoader |
|   | hadoop.crypto.TestCryptoCodec |
|   | hadoop.yarn.server.resourcemanager.scheduler.capacity.TestCapacitySchedulerMultiNodes |
|   | hadoop.yarn.client.api.impl.TestOpportunisticContainerAllocationE2E |
|   | hadoop.yarn.client.api.impl.TestAMRMProxy |
|   | hadoop.yarn.client.api.impl.TestYarnClient |
|   | hadoop.yarn.client.api.impl.TestNMClient |
|   | hadoop.yarn.client.api.impl.TestAMRMClientPlacementConstraints |
|   | hadoop.yarn.client.api.impl.TestAMRMClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.48 ServerAPI=1.48 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7608 |
| Optional Tests | dupname asflicense codespell detsecrets hadolint shellcheck shelldocs jsonlint compile cc mvnsite javac unit golang javadoc mvninstall shadedclient xmllint |
| uname | Linux ed869f1b88db 5.15.0-131-generic #141-Ubuntu SMP Fri Jan 10 21:18:28 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 877f79282a0789d17255605b1b3ee94b5cf269c5 |
| Default Java | Red Hat, Inc.-1.8.0_412-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/testReport/ |
| Max. process+thread count | 2370 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-hdfs-project/hadoop-hdfs-native-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/1/console |
| versions | git=2.9.5 maven=3.6.3 xmllint=20901 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

vinayakumarb commented:
All the failures in compilation are due to below. Nothing to do with changes in this PR.
```
[ERROR] PROTOC FAILED: /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe)
/home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe)
/home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe)
--grpc-java_out: protoc-gen-grpc-java: Plugin failed with status code 1.

[ERROR] /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/src/main/proto/csi.proto [0:0]: /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe)
/home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe)
/home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /home/jenkins/jenkins-agent/workspace/hadoop-multibranch_PR-7608/centos-7/src/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi/target/protoc-plugins/protoc-gen-grpc-java-1.69.0-linux-x86_64.exe)
--grpc-java_out: protoc-gen-grpc-java: Plugin failed with status code 1.
```

vinayakumarb commented:
> The error message on aarch64 ain't cheese
> 
> ```
> ayushsaxena@ayushsaxena hadoop % ./start-build-env.sh ubuntu_24 
> [+] Building 0.0s (2/2) FINISHED                                                                                                                                                              
>  => [internal] load .dockerignore                                                                                                                                                        0.0s
>  => => transferring context: 2B                                                                                                                                                          0.0s
>  => [internal] load build definition from Dockerfile_ubuntu_24_aarch64                                                                                                                   0.0s
>  => => transferring dockerfile: 2B                                                                                                                                                       0.0s
> ERROR: failed to solve: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount1675973539/Dockerfile_ubuntu_24_aarch64: no such file or directory
> ayushsaxena@ayushsaxena hadoop % 
> ```
> 
> Can we update the BUILDING.txt to reflect which ones are supported for aarch64.
> 
> Rest if the build comes clean, changes LGTM

Ack. Updated the BUILDING.txt and added error message in `start-build-env.sh` as well.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 52s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  hadolint  |   0m  1s |  |  hadolint was not available.  |
| +0 :ok: |  shellcheck  |   0m  1s |  |  Shellcheck was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +0 :ok: |  jsonlint  |   0m  1s |  |  jsonlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 51s |  |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |  37m  7s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/branch-mvninstall-root.txt) |  root in trunk failed.  |
| -1 :x: |  compile  |  12m 43s | [/branch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/branch-compile-root.txt) |  root in trunk failed.  |
| -1 :x: |  mvnsite  |   6m 15s | [/branch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/branch-mvnsite-root.txt) |  root in trunk failed.  |
| -1 :x: |  javadoc  |   7m 29s | [/branch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/branch-javadoc-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  shadedclient  |  46m 17s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |  36m 13s | [/patch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/patch-mvninstall-root.txt) |  root in the patch failed.  |
| -1 :x: |  compile  |  13m 10s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  cc  |  13m 10s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  golang  |  13m 10s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  javac  |  13m 10s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -1 :x: |  mvnsite  |   5m  3s | [/patch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/patch-mvnsite-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  xmllint  |   0m  1s |  |  No new issues.  |
| -1 :x: |  javadoc  |   6m 21s | [/patch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/patch-javadoc-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  shadedclient  |  46m  6s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 507m 18s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| -1 :x: |  asflicense  |   1m  9s | [/results-asflicense.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/results-asflicense.txt) |  The patch generated 1 ASF License warnings.  |
|  |   | 722m 33s |  |  |


| Reason | Tests |
|-------:|:------|
| Unreaped Processes | root:2 |
| Failed junit tests | hadoop.util.TestNativeCodeLoader |
|   | hadoop.crypto.TestCryptoCodec |
|   | hadoop.yarn.server.resourcemanager.scheduler.capacity.TestCapacitySchedulerMultiNodes |
|   | hadoop.yarn.client.api.impl.TestOpportunisticContainerAllocationE2E |
|   | hadoop.yarn.client.api.impl.TestAMRMProxy |
|   | hadoop.yarn.client.api.impl.TestYarnClient |
|   | hadoop.yarn.client.api.impl.TestNMClient |
|   | hadoop.yarn.client.api.impl.TestAMRMClientPlacementConstraints |
|   | hadoop.yarn.client.api.impl.TestAMRMClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.48 ServerAPI=1.48 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7608 |
| Optional Tests | dupname asflicense codespell detsecrets hadolint shellcheck shelldocs jsonlint compile cc mvnsite javac unit golang javadoc mvninstall shadedclient xmllint |
| uname | Linux 4679aca24879 5.15.0-131-generic #141-Ubuntu SMP Fri Jan 10 21:18:28 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 877f79282a0789d17255605b1b3ee94b5cf269c5 |
| Default Java | Red Hat, Inc.-1.8.0_412-b08 |
| Unreaped Processes Log | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/artifact/out/patch-unit-root-reaper.txt |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/testReport/ |
| Max. process+thread count | 2609 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-hdfs-project/hadoop-hdfs-native-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/2/console |
| versions | git=2.9.5 maven=3.6.3 xmllint=20901 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

ayushtkn reviewed (approved):
LGTM.

Something crazy with the last build as well, see if you can get a clean build, before merging, haven't seen this error before, not sure why is it crying in yarn for protoc

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  43m 32s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  hadolint  |   0m  1s |  |  hadolint was not available.  |
| +0 :ok: |  shellcheck  |   0m  1s |  |  Shellcheck was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +0 :ok: |  jsonlint  |   0m  1s |  |  jsonlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 26s |  |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |  21m 13s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/branch-mvninstall-root.txt) |  root in trunk failed.  |
| -1 :x: |  compile  |   6m 18s | [/branch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/branch-compile-root.txt) |  root in trunk failed.  |
| -1 :x: |  mvnsite  |   2m 33s | [/branch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/branch-mvnsite-root.txt) |  root in trunk failed.  |
| -1 :x: |  javadoc  |   4m 30s | [/branch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/branch-javadoc-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  shadedclient  |  27m  2s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |  22m 36s | [/patch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/patch-mvninstall-root.txt) |  root in the patch failed.  |
| -1 :x: |  compile  |   7m  5s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  cc  |   7m  5s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  golang  |   7m  5s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  javac  |   7m  5s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -1 :x: |  mvnsite  |   2m 53s | [/patch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/patch-mvnsite-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| -1 :x: |  javadoc  |   4m  8s | [/patch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/patch-javadoc-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  shadedclient  |  25m 37s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 483m 11s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 48s |  |  The patch does not generate ASF License warnings.  |
|  |   | 653m  8s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.util.TestNativeCodeLoader |
|   | hadoop.crypto.TestCryptoCodec |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.48 ServerAPI=1.48 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7608 |
| Optional Tests | dupname asflicense codespell detsecrets hadolint shellcheck shelldocs jsonlint compile cc mvnsite javac unit golang javadoc mvninstall shadedclient xmllint |
| uname | Linux 472735767fd0 5.15.0-136-generic #147-Ubuntu SMP Sat Mar 15 15:53:30 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 0db112278bb38edeb7787a15f2a4aae531d4ff1d |
| Default Java | Red Hat, Inc.-1.8.0_412-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/testReport/ |
| Max. process+thread count | 3299 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-hdfs-project/hadoop-hdfs-native-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/3/console |
| versions | git=2.9.5 maven=3.6.3 xmllint=20901 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

vinayakumarb commented:
Ran the failed tests locally.. 
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.hadoop.crypto.TestCryptoCodec
[WARNING] Tests run: 5, Failures: 0, Errors: 0, Skipped: 3, Time elapsed: 21.64 s - in org.apache.hadoop.crypto.TestCryptoCodec
[INFO] Running org.apache.hadoop.util.TestNativeCodeLoader
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.215 s - in org.apache.hadoop.util.TestNativeCodeLoader
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 6, Failures: 0, Errors: 0, Skipped: 3
[INFO] 
```

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  41m 11s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  hadolint  |   0m  0s |  |  hadolint was not available.  |
| +0 :ok: |  shellcheck  |   0m  0s |  |  Shellcheck was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +0 :ok: |  jsonlint  |   0m  0s |  |  jsonlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   7m 16s |  |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |  31m 40s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/branch-mvninstall-root.txt) |  root in trunk failed.  |
| -1 :x: |  compile  |  11m 33s | [/branch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/branch-compile-root.txt) |  root in trunk failed.  |
| -1 :x: |  mvnsite  |   4m 38s | [/branch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/branch-mvnsite-root.txt) |  root in trunk failed.  |
| -1 :x: |  javadoc  |   6m 36s | [/branch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/branch-javadoc-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  shadedclient  |  39m  2s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |  30m 24s | [/patch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/patch-mvninstall-root.txt) |  root in the patch failed.  |
| -1 :x: |  compile  |  10m 44s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  cc  |  10m 44s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  golang  |  10m 44s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  javac  |  10m 44s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -1 :x: |  mvnsite  |   4m 41s | [/patch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/patch-mvnsite-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| -1 :x: |  javadoc  |   6m 24s | [/patch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/patch-javadoc-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  shadedclient  |  40m 45s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 504m  2s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| -1 :x: |  asflicense  |   1m  1s | [/results-asflicense.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/results-asflicense.txt) |  The patch generated 1 ASF License warnings.  |
|  |   | 731m 14s |  |  |


| Reason | Tests |
|-------:|:------|
| Unreaped Processes | root:2 |
| Failed junit tests | hadoop.fs.TestHdfsNativeCodeLoader |
|   | hadoop.crypto.TestCryptoCodec |
|   | hadoop.util.TestNativeCodeLoader |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.49 ServerAPI=1.49 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7608 |
| Optional Tests | dupname asflicense codespell detsecrets hadolint shellcheck shelldocs jsonlint compile cc mvnsite javac unit golang javadoc mvninstall shadedclient xmllint |
| uname | Linux 3c1e7344c191 5.15.0-131-generic #141-Ubuntu SMP Fri Jan 10 21:18:28 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / bcd73e6dfcc32b9ee99d537af486362bf82451ad |
| Default Java | Red Hat, Inc.-1.8.0_412-b08 |
| Unreaped Processes Log | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/artifact/out/patch-unit-root-reaper.txt |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/testReport/ |
| Max. process+thread count | 3572 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-hdfs-project/hadoop-hdfs-native-client . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/4/console |
| versions | git=2.9.5 maven=3.6.3 xmllint=20901 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

pan3793 commented on BUILDING.txt:
```diff
@@ -29,7 +29,11 @@ This requires a recent version of docker (1.4.1 and higher are known to work).
+
+   - [OS Platform] One of [centos_7, centos_8, debian_10, ubuntu_20, ubuntu_24, windows_10].
+        Default is 'ubuntu_20'.
+        Note: Currently only  default ('ubuntu_20') is supported on arm machine
```
nit: remove an extra space before "default"

Just out of curiosity, have you encountered any issues with supporting ubuntu_24 on ARM, the support might be worth doing, given that a lot of developers use Apple Silicon

vinayakumarb replied:
Supporting ARM in ubuntu-22+ needs some amount of effort. Ex: `phantomjs` library is not available for arm in recent OS distributions.

I will try to address that in a follow up PR.

Thanks for checking.

pan3793 replied:
SGTM, thank you for sharing that.

pan3793 commented on hadoop-project/pom.xml:
```diff
@@ -2747,6 +2747,16 @@
       </dependencyManagement>
     </profile>
+    <profile>
+      <id>jdk-17</id>
```
can we follow the [apache/maven-apache-parent](https://github.com/apache/maven-apache-parent/blob/2591ca2d83b71d18a621d01621f078f25cfd456e/pom.xml#L501) to name it `jdk17+`

vinayakumarb replied:
Done.

pan3793 commented on hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/include/hdfspp/block_location.h:
```diff
@@ -19,6 +19,8 @@
 #ifndef HDFSPP_BLOCK_LOCATION_H
 #define HDFSPP_BLOCK_LOCATION_H
 
+#include <stdint.h>
```
there is a dedicated PR that just merged to fix it, https://github.com/apache/hadoop/issues/6191.

vinayakumarb replied:
Thanks for pointing this. Have removed related changes from this PR.

cnauroth reviewed (approved):
+1 from me, pending resolution of the last round of feedback from @pan3793 . Thanks @vinayakumarb !

vinayakumarb committed:
HADOOP-19540. [JDK17] Add ubuntu:noble as a build platform with JDK-17 as default

1. Added "ubuntu:noble" to supported docker platforms.
2. Modified `start-build-env.sh` to use expected Dockerfiles dynamically.
   ex: `bash start-build-env.sh ubuntu_24` where `ubuntu_24` is the extra suffix in the name of the Dockerfile.
3. Added `jdk17++` profile activated when jdk17 is available.

Successfully built the entire Hadoop tar with native support on env created using `bash start-build-env.sh ubuntu_24`

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  46m 15s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  hadolint  |   0m  0s |  |  hadolint was not available.  |
| +0 :ok: |  shellcheck  |   0m  0s |  |  Shellcheck was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +0 :ok: |  jsonlint  |   0m  0s |  |  jsonlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   6m 36s |  |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |  36m 56s | [/branch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/branch-mvninstall-root.txt) |  root in trunk failed.  |
| -1 :x: |  compile  |  12m 59s | [/branch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/branch-compile-root.txt) |  root in trunk failed.  |
| -1 :x: |  mvnsite  |   5m 21s | [/branch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/branch-mvnsite-root.txt) |  root in trunk failed.  |
| -1 :x: |  javadoc  |   6m 42s | [/branch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/branch-javadoc-root.txt) |  root in trunk failed.  |
| +1 :green_heart: |  shadedclient  |  45m 13s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 33s |  |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |  36m 32s | [/patch-mvninstall-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/patch-mvninstall-root.txt) |  root in the patch failed.  |
| -1 :x: |  compile  |  12m 54s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| -1 :x: |  javac  |  12m 54s | [/patch-compile-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/patch-compile-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -1 :x: |  mvnsite  |   4m 53s | [/patch-mvnsite-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/patch-mvnsite-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| -1 :x: |  javadoc  |   6m 26s | [/patch-javadoc-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/patch-javadoc-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  shadedclient  |  45m 56s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 913m  3s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/patch-unit-root.txt) |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 52s |  |  The patch does not generate ASF License warnings.  |
|  |   | 1172m 52s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.sls.appmaster.TestAMSimulator |
|   | hadoop.mapred.gridmix.TestGridmixSubmission |
|   | hadoop.mapred.gridmix.TestLoadJob |
|   | hadoop.mapred.gridmix.TestSleepJob |
|   | hadoop.fs.s3a.prefetch.TestS3ACachingBlockManager |
|   | hadoop.util.TestNativeCodeLoader |
|   | hadoop.crypto.TestCryptoCodec |
|   | hadoop.hdfs.server.namenode.TestCacheDirectives |
|   | hadoop.fs.viewfs.TestViewFileSystemOverloadSchemeHdfsFileSystemContract |
|   | hadoop.hdfs.TestDatanodeRegistration |
|   | hadoop.hdfs.server.blockmanagement.TestBlockReportLease |
|   | hadoop.fs.TestFcHdfsCreateMkdir |
|   | hadoop.fs.TestHDFSFileContextMainOperations |
|   | hadoop.fs.viewfs.TestViewFileSystemLinkFallback |
|   | hadoop.fs.viewfs.TestViewFileSystemLinkMergeSlash |
|   | hadoop.hdfs.server.blockmanagement.TestBlockTokenWithShortCircuitRead |
|   | hadoop.fs.TestHdfsNativeCodeLoader |
|   | hadoop.fs.viewfs.TestViewFsAtHdfsRoot |
|   | hadoop.hdfs.server.blockmanagement.TestDatanodeManager |
|   | hadoop.hdfs.server.namenode.TestCacheDirectivesWithViewDFS |
|   | hadoop.fs.TestSWebHdfsFileContextMainOperations |
|   | hadoop.fs.viewfs.TestViewFileSystemHdfs |
|   | hadoop.fs.viewfs.TestViewFileSystemLinkRegex |
|   | hadoop.fs.viewfs.TestViewFileSystemAtHdfsRoot |
|   | hadoop.fs.viewfs.TestViewFsHdfs |
|   | hadoop.fs.TestWebHdfsFileContextMainOperations |
|   | hadoop.yarn.applications.distributedshell.TestDSTimelineV10 |
|   | hadoop.yarn.applications.distributedshell.TestDSWithMultipleNodeManager |
|   | hadoop.yarn.applications.distributedshell.TestDSTimelineV20 |
|   | hadoop.yarn.applications.distributedshell.TestDSTimelineV15 |
|   | hadoop.yarn.server.resourcemanager.scheduler.capacity.TestCapacitySchedulerMultiNodes |
|   | hadoop.yarn.server.router.clientrm.TestFederationClientInterceptor |
|   | hadoop.yarn.server.router.TestRouterMetrics |
|   | hadoop.yarn.server.router.secure.TestSecureLogins |
|   | hadoop.yarn.server.router.webapp.TestFederationInterceptorREST |
|   | hadoop.yarn.server.router.TestRouterServerUtil |
|   | hadoop.yarn.server.router.webapp.TestFederationInterceptorRESTRetry |
|   | hadoop.yarn.client.api.impl.TestOpportunisticContainerAllocationE2E |
|   | hadoop.yarn.client.api.impl.TestAMRMProxy |
|   | hadoop.yarn.client.api.impl.TestYarnClient |
|   | hadoop.yarn.client.api.impl.TestNMClient |
|   | hadoop.yarn.client.api.impl.TestAMRMClientPlacementConstraints |
|   | hadoop.yarn.client.api.impl.TestAMRMClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.49 ServerAPI=1.49 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/7608 |
| Optional Tests | dupname asflicense codespell detsecrets hadolint shellcheck shelldocs jsonlint compile javac javadoc mvninstall mvnsite unit shadedclient xmllint |
| uname | Linux 396c4e0b03ef 5.15.0-131-generic #141-Ubuntu SMP Fri Jan 10 21:18:28 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 1bea33ca1baceb09812865c738c4f52c956b477f |
| Default Java | Red Hat, Inc.-1.8.0_412-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/testReport/ |
| Max. process+thread count | 2778 (vs. ulimit of 5500) |
| modules | C: hadoop-project . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-7608/5/console |
| versions | git=2.9.5 maven=3.6.3 xmllint=20901 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

vinayakumarb commented:
Thanks everyone for reviews.
Mergiing now.
All the failures shown above are not related at all.

slfan1989 commented:
> Thanks everyone for reviews.
> Mergiing now.
> All the failures shown above are not related at all.

@vinayakumarb Thank you for the contribution!

slfan1989 commented:
@vinayakumarb Should the PR supporting Ubuntu 24 be backported to branch-3.4, even though branch-3.4 currently does not support JDK17?

cc: @steveloughran

steveloughran commented:
no, this shouldn't have been backported to branch-3.R as it is breaking releases there.

steveloughran commented:
update, I'm confused, tried to build on the wrong branch. sorry. I'd actually kicked off a build of trunk in an arm64 ec2 host...if it failed its because the docker image wasn't java17 everywhere...
